xrootd/protocol/signRequirements.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package protocol contains the XRootD protocol specific types.
+package protocol // import "go-hep.org/x/hep/xrootd/protocol"
+
+import (
+	"go-hep.org/x/hep/xrootd/protocol/protocol"
+)
+
+// SignRequirements implements a way to check if request should be signed
+// according to XRootD protocol specification v. 3.1.0, p.75-76.
+type SignRequirements struct {
+	requirements map[uint16]protocol.RequestLevel
+}
+
+// Needed checks if the request should be signed.
```

Comment:
`s/Needed checks if the request should be signed./Needed returns whether the request should be signed./`

`modifies` should also probably be documented.